### PR TITLE
Refactor signAndPost + add a few tests

### PR DESCRIPTION
**Pull request by erik-source**
Move out signAndPost into it's own file.
Move out signAndPost tests into their own file.

Co-authored-by: Theo Kalfas <tkalfas@sourceallies.com>

### CODE REVIEW

File: test/SignAndPost.test.js [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+      result = underTest.doSignAndPost(
+        chance.string(),
+        chance.string(),
+        chance.string(),
+        chance.string(),
+        chance.string(),
+      );
+    });
+
+    it("throws an error when there is an error in the OAuth post response", async () => {
```
Comment (nbodesa) [inline test/SignAndPost.test.js:202]:
```suggestion
    it("should reject with the same error", async () => {
```

File: test/SignAndPost.test.js [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+      result = underTest.doSignAndPost(
+        chance.string(),
+        chance.string(),
+        chance.string(),
+        chance.string(),
+        chance.string(),
+      );
+    });
+
+    it("should return an error", async () => {
```
Comment (nbodesa) [inline test/SignAndPost.test.js:242]:
```suggestion
    it("should return the correct status text", async () => {
```

File: test/SignAndPost.test.js [hunk 3]
```diff
@@ -0,0 +1,383 @@
...
```
Comment (nbodesa) [inline test/SignAndPost.test.js:282]:
```suggestion
    it("should return the correct status text", async () => {
```

File: test/SignAndPost.test.js [hunk 4]
```diff
@@ -0,0 +1,383 @@
...
+      result = underTest.doSignAndPost(
+        chance.string(),
+        chance.string(),
+        chance.string(),
+        chance.string(),
+        chance.string(),
+      );
+    });
+
+    it("returns the same response", async () => {
```
Comment (nbodesa) [inline test/SignAndPost.test.js:319]:
```suggestion
    it("should return the same response", async () => {
```

File: test/SignAndPost.test.js [hunk 5]
```diff
@@ -0,0 +1,383 @@
...
+      await underTest.doSignAndPost(
+        expectedLinkToOpen,
+        expectedAccessToken,
+        expectedAccesssTokenSecret,
+        expectedPostBody,
+        expectedPostBodyContentType
+      );
+    });
+
+    it("should pass parameters to oauthSession", () => {
```
Comment (nbodesa) [inline test/SignAndPost.test.js:375]: Split these up as separate its like @blarmon  suggested in the other PR. I like that idea

File: test/SignAndPost.test.js [hunk 6]
```diff
@@ -0,0 +1,383 @@
...
+      await underTest.doSignAndPost(
+        chance.string(),
+        chance.string(),
+        chance.string(),
+        chance.string(),
+        chance.string()
+      );
+    });
+
+    it("creates an OAuth with the custom JD Accept header", () => {
```
Comment (blarmon) [inline test/SignAndPost.test.js:128]:
```suggestion
    it("should create an OAuth with the custom JD Accept header", () => {
```

File: test/SignAndPost.test.js [hunk 7]
```diff
@@ -0,0 +1,383 @@
...
+        chance.string(),
+        chance.string(),
+        chance.string(),
+        chance.string(),
+        chance.string(),
+        expectedAuthorizeCallbackUri
+      );
+    });
+
+    it("creates an OAuth with the provided optionalAuthorizationCallbackUri", () => {
```
Comment (blarmon) [inline test/SignAndPost.test.js:159]:
```suggestion
    it("should create an OAuth with the provided optionalAuthorizationCallbackUri", () => {
```

File: test/SignAndPost.test.js (hunk 5 above)
Comment (erik-source) [reply test/SignAndPost.test.js]: Updated :)

**Pull request merged**